core/integration/file_test.go

```diff
@@ -192,3 +195,46 @@ func TestFileWithTimestamps(t *testing.T) {
 	require.Contains(t, ls, "Access: 1985-10-26 08:15:00.000000000 +0000")
 	require.Contains(t, ls, "Modify: 1985-10-26 08:15:00.000000000 +0000")
 }
+
+func TestFileContents(t *testing.T) {
+	c, ctx := connect(t)
+	defer c.Close()
+
+	src := c.Host().Directory(".")
+	alpine := c.Container().
+		From("alpine:3.16.2").
+		WithDirectory("/src", src).WithWorkdir("/src")
+
+	// Set three types of file sizes for test data,
+	// the third one uses a size larger than the max chunk size:
+	fileSizes := []int{
+		// core.MaxFileContentsChunkSize / 2,
+		// core.MaxFileContentsChunkSize,
+		core.MaxFileContentsChunkSize * 2,
+	}
+
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+
+	// Generate and write data for each file size:
+	for i, size := range fileSizes {
+		filename := strconv.Itoa(i)
+		dest := filepath.Join(wd, filename)
+		var buf bytes.Buffer
+		for i := 0; i < size; i++ {
+			buf.WriteByte('a')
+		}
+		err := os.WriteFile(dest, buf.Bytes(), 0600)
```

Comment:
I get the approach you were going for here and like the idea, but unfortunately due to the way local directory syncs work I don't think it quite works out.

Local directory syncs aren't like bind mounts (e.g. `-v` flag in `docker run`) where any change you make to your working dir will automatically also result in a change in the container.

Instead, the contents of `c.Host().Directory(".")` are synced into the engine once, at the first time you use it. This happens once per session (i.e. per `dagger.Connect` call).

So I think you can update the test to instead pre-create all the files and then load them into the container.

Reply:
@sipsma Good to know, added a fix in 21a9522306a194a4af5324e6faf72099b7b5ed5e